testing/e2e/suites/osm.go

```diff
@@ -0,0 +1,64 @@
+package suites
+
+import (
+	"context"
+
+	"github.com/Azure/aks-app-routing-operator/testing/e2e/infra"
+	"github.com/Azure/aks-app-routing-operator/testing/e2e/manifests"
+	corev1 "k8s.io/api/core/v1"
+	netv1 "k8s.io/api/networking/v1"
+	"k8s.io/client-go/rest"
+)
+
+var (
+	osmNs = make(map[string]*corev1.Namespace)
+)
+
+func osmSuite(in infra.Provisioned) []test {
+	return []test{
+		{
+			name: "osm ingress",
+			cfgs: builderFromInfra(in).
+				withOsm(in, true).
+				withVersions(manifests.AllOperatorVersions...).
+				withZones(manifests.AllDnsZoneCounts, manifests.AllDnsZoneCounts).build(),
+			run: func(ctx context.Context, config *rest.Config, operator manifests.OperatorConfig) error {
+				if err := clientServerTest(ctx, config, operator, osmNs, in,
+					func(ingress *netv1.Ingress, service *corev1.Service, _ zoner) error {
+						annotations := ingress.GetAnnotations()
+						annotations["kubernetes.azure.com/use-osm-mtls"] = "true"
+						ingress.SetAnnotations(annotations)
+
+						return nil
+					}); err != nil {
+					return err
+				}
+
+				return nil
+			},
+		},
+		{
+			name: "osm service",
```

Comment:
will we remove this when we fully shift to expected behavior where the client should create their own ingress? don't wanna have too much bloat in the e2e suite

Reply:
sure. We have to have tests for it until then.